src/main/java/microsoft/sql/Vector.java

```diff
@@ -0,0 +1,327 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+
+package microsoft.sql;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.Arrays;
+import java.util.ResourceBundle;
+import java.text.MessageFormat;
+
+public final class Vector implements java.io.Serializable {
+    public enum VectorDimensionType {
+        float16, // 16-bit (half precision) float

```

Comment:
Do we even want to expose float16, yet? What are the implications if we expose it before the server supports it? What if we only add/expose it later? (Don't remove code yet, just want to have the conversation before making a decision.)